python/pip_install/pip_repository.bzl

```diff
@@ -257,110 +257,143 @@ A requirements_lock attribute must be specified, or a platform-specific lockfile
 """)
     return requirements_txt
 
-# Keep in sync with `_clean_name` in generated requirements.bzl
+# Keep in sync with `_clean_pkg_name` in generated bzlmod requirements.bzl
 def _clean_pkg_name(name):
     return name.replace("-", "_").replace(".", "_").lower()
 
-def _bzlmod_pkg_aliases(rctx, requirements_txt):
+def _bzlmod_pkg_aliases(repo_name, bzl_packages):
     """Create alias declarations for each python dependency.
 
-    The aliases should be appended to the pip_parse repo's BUILD.bazel file. These aliases
+    The aliases should be appended to the pip_repository BUILD.bazel file. These aliases
     allow users to use requirement() without needed a corresponding `use_repo()` for each dep
     when using bzlmod.
 
     Args:
-        rctx: the repository context
-        requirements_txt: label to the requirements lock file
+        repo_name: the repository name of the parent that is visible to the users.
+        bzl_packages: the list of packages to setup.
     """
-    requirements = parse_requirements(rctx.read(requirements_txt)).requirements
-
     build_content = ""
-    for requirement in requirements:
+    for name in bzl_packages:
         build_content += """\
 
 alias(
     name = "{name}_pkg",
-    actual = "@{repo_prefix}{dep}//:pkg",
+    actual = "@{repo_name}_{dep}//:pkg",
 )
 
 alias(
     name = "{name}_whl",
-    actual = "@{repo_prefix}{dep}//:whl",
+    actual = "@{repo_name}_{dep}//:whl",
 )
 
 alias(
     name = "{name}_data",
-    actual = "@{repo_prefix}{dep}//:data",
+    actual = "@{repo_name}_{dep}//:data",
 )
 
 alias(
     name = "{name}_dist_info",
-    actual = "@{repo_prefix}{dep}//:dist_info",
+    actual = "@{repo_name}_{dep}//:dist_info",
 )
 """.format(
-            name = _clean_pkg_name(requirement[0]),
-            repo_prefix = rctx.attr.repo_prefix,
-            dep = _clean_pkg_name(requirement[0]),
+            name = name,
+            repo_name = repo_name,
+            dep = name,
         )
 
     return build_content
 
 def _pip_repository_impl(rctx):
-    python_interpreter = _resolve_python_interpreter(rctx)
-
-    # Write the annotations file to pass to the wheel maker
-    annotations = {package: json.decode(data) for (package, data) in rctx.attr.annotations.items()}
-    annotations_file = rctx.path("annotations.json")
-    rctx.file(annotations_file, json.encode_indent(annotations, indent = " " * 4))
-
     requirements_txt = locked_requirements_label(rctx, rctx.attr)
-    args = [
-        python_interpreter,
-        "-m",
-        "python.pip_install.tools.lock_file_generator.lock_file_generator",
```

Comment:
Does this mean lock_file_generator can be deleted, too?

Reply:
Done.